iree/turbine/kernel/wave/index_sequence_analysis.py

```diff
@@ -125,10 +113,135 @@ def has_strided_access(node: fx.Node) -> bool:
                     )
                     for j, dim in enumerate(custom.register_type.symbolic_shape)
                 }
+                ops_to_combine.append(write)
 
+        # Useful to handle write/read dependency
+        custom.replace_all_uses_with(ops_to_combine)
         custom.graph.erase_node(operator)
 
 
+def partition_ops_with_gpr_offsets(trace: CapturedTrace, constraints: list[Constraint]):
+    """
+    This function analyzes the index sequence of reads and writes in a graph.
+    If the reads or writes have incontiguous offsets based on GPR_NUM, we'd
+    need to split these reads/writes appropriately.
+
+    e.g a vector<16xf16> may be owned by lane 0, and lane 16 in this layout:
+    [0, 0, 0, 0, 16, 16, 16, 16, 0, 0, 0, 0, 16, 16, 16, 16].
+
+    With our current gloassary, this means we have 2 VGPR "chunks".
+    [0:4) and [8:12) for lane0, and [4:8) and [12:16) for lane16.
+    To the lane it should just look like vector<8xf16>.
+    Hence for this example, we'd need two reads of vector<4xf16> and a couple
+    insert_slices to combine them together into a single vector<8xf16>.
+
+    """
+
+    def has_gpr_offsets(node: fx.Node) -> bool:
+        """
+        Checks for writes on 2d tensors with strided access on a single dimension that
+        read more than a single element.
+        """
+        custom = get_custom(node)
+        if not isinstance(custom, (Read, Write)):
+            return False
+        dims_with_gpr_offset = [
```

Comment:
done(kind of), thanks! I modified it to check if its 1 or 0, since with the case above we will run into notimplemented error on Read/Writes with no GPR offset at all. LMK if it sounds good to you